lib/utils/job-utils/redfish-tool.js

```diff
@@ -50,105 +52,54 @@ function redfishToolFactory(
-            });
 
-            request.on('error', function(error) {
-                reject(error);
-            });
-            
-            switch(options.method) {
-            case 'POST':
-            case 'PATCH':
-            case 'PUT':
-                request.write(data);
-                break;
-            default: 
-                break;
-            }
-            request.end();
-        });
-    }); 
-    
     /**
      * @function clientRequest
      * @description make request to HTTP client
      * @param path the URI path to send the request to
      * @param method the HTTP method: GET,POST,PUT,DELETE,PATCH. Default: GET
      * @param data the POST/PUT/PATCH data to write to the HTTP client
      */  
-    RedfishTool.prototype.clientRequest = function (path, method, data) {
+
+    RedfishTool.prototype.clientRequest = function(path, method, data) {
         var self = this;
-        var protocol = self.settings.protocol || 'http';
-        var options = {};
-        var headers = {
-            'Content-Type': 'application/json'
-        };
-        data = data || '';
-        var length = data.length;
-        if (0 !== length) {       
-            data = JSON.stringify(data);
-            headers['Content-Length'] = Buffer.byteLength(data);
-        }
-        options.headers = headers;
-        options.method = method || 'GET';
-        options.host = self.settings.host;
-        options.port = self.settings.port;
-        options.path = path || self.settings.root || '/';
-        options.rejectUnauthorized = self.settings.verifySSL || false;
-        options.auth = [ self.settings.username, self.settings.password ].join(':');
-        options.recvTimeoutMs = self.settings.recvTimeoutMs;
-        return self.request(options, protocol, data)
-        .then(function(response) {
-            if (response.httpStatusCode > 206) {
-                logger.error('HTTP Error', response);
-                var errorMsg = _.get(response, 'body.error.message', 'Unknown Error');
-                throw new Error(errorMsg);
-            }
-            if (response.body.length > 0) {
-                response.body = JSON.parse(response.body);
-            }
-            return response;
+        var setups = {};
+
+        setups.url = {};
+        setups.url.protocol = self.settings.protocol || 'http';
+        setups.url.host = self.settings.host;
+        setups.url.port = self.settings.port;
+        setups.url.path = path || self.settings.root || '/';
+
+        setups.method = method || 'GET';
+        setups.credential = {username: self.settings.username, 
+            password: self.settings.password};
+        setups.verifySSl = self.settings.verifySSl || false;
+        setups.headers = {'Content-Type': 'application/json'};
+        setups.recvTimeoutMs = self.settings.recvTimeoutMs;
+        setups.data = data;
+
+        var http = new HttpTool();
+        
+        return new Promise(function(resolve){
```

Comment:
You don't need to create a new Promise here, since `http.setupRequest` has already returned promise, you should could just return the promise chain.
If you need to promise chain resolves `responce`, you just need to change line#98 to `return response`